### PR TITLE
TEST: Don't merge: check all args are Basic

### PR DESCRIPTION
**Pull request by oscarbenjamin**
#### References to other Issues or PRs

See e.g. https://github.com/sympy/sympy/pull/20635#discussion_r546786841

#### Brief description of what is fixed or changed

This is just an experiment to flush out any classes that have non-Basic instances in their args.

#### Release Notes

NO ENTRY

### CODE REVIEW

**Comment by asmeurer** [comment]
Is [this](https://github.com/sympy/sympy/pull/20641/files#diff-8362722f7406300388c979bff45233f8eb6cae1be6ae44bc503d20d8d9c371b6) something that it just started doing on its own or did we configure this? It doesn't seem useful in its current state but maybe it could be fixed so it is useful.

**Comment by oscarbenjamin** [comment]
It's just something that github started doing. I'm not sure how it works.

File: sympy/combinatorics/graycode.py [hunk 1]
```diff
@@ -85,6 +86,7 @@ def __new__(cls, n, *args, **kw_args):
                 'Gray code dimension must be a positive integer, not %i' % n)
         n = int(n)
         args = (n,) + args
+        args = (_sympify(arg) for arg in args)
```
Comment (smichr) [inline sympy/combinatorics/graycode.py:89]: This makes me worry that any calcs that assume args are ints will now be slowed down a lot. This was something we had to be careful of in the ntheory/prime work.

Comment (oscarbenjamin) [reply sympy/combinatorics/graycode.py:89]: To be clear I don't intend for this to be merged.

File: sympy/core/basic.py [hunk 2]
```diff
@@ -1921,7 +1932,7 @@ def _atomic(e, recursive=False):
     seen = set()
     if isinstance(e, Basic):
         free = getattr(e, "free_symbols", None)
-        if free is None:
```
Comment (smichr) [inline sympy/core/basic.py:1924]: None was the default for an object that didn't have the `free_symbols` attribute. You are now returning `{e}` if `e` is a SymPy object with no free symbols.

Comment (oscarbenjamin) [reply sympy/core/basic.py:1924]: I'm actually not sure where all these changes came from. I had these commits sitting around and just pushed them to see what would happen with the tests.

**Pull request closed without merging**